File: slingcessories_mobile_flutter/lib/models/create_accessory_dto.dart
class CreateAccessoryDto {
  final String title;
  final String? pictureUrl;
  final double price;
  final String? url;
  final bool wishlist;
  final int categoryId;
  final int? subcategoryId;
  final Map<String, int> slinghotQuantities;

  CreateAccessoryDto({
    required this.title,
    this.pictureUrl,
    required this.price,
    this.url,
    required this.wishlist,
    required this.categoryId,
    this.subcategoryId,
    required this.slinghotQuantities,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'pictureUrl': pictureUrl,
      'price': price,
      'url': url,
      'wishlist': wishlist,
      'categoryId': categoryId,
      'subcategoryId': subcategoryId,
      'slinghotQuantities': slinghotQuantities,
    };
  }
}

File: Slingcessories.Mobile.Maui/Models/CreateAccessoryDto.cs
namespace Slingcessories.Mobile.Maui.Models;

public class CreateAccessoryDto
{
    public string Title { get; set; } = string.Empty;
    public string? PictureUrl { get; set; }
    public decimal Price { get; set; }
    public string? Url { get; set; }
    public bool Wishlist { get; set; }
    public int CategoryId { get; set; }
    public int? SubcategoryId { get; set; }
    public Dictionary<int, int>? SlinghotQuantities { get; set; }
}
